### PR TITLE
SOLVE issue-Add an emoji library to add emojis in a message that we c…

### PR DESCRIPTION
**Pull request by nishant25062002**
…an send #56

### CODE REVIEW

File: package-lock.json
Comment (mdshamoon) [inline package-lock.json:1]: We are using yarn for development. Please remove this file.

File: package.json [hunk 1]
```diff
@@ -27,13 +27,14 @@
...
     "react-native-dotenv": "^3.4.8",
+    "react-native-emoji-picker": "^0.2.2",
+    "react-native-emoji-selector": "^0.2.0",
     "react-native-gesture-handler": "~2.9.0",
     "react-native-pager-view": "6.1.2",
     "react-native-phone-number-input": "^2.1.0",
     "react-native-reanimated": "~2.14.4",
     "react-native-safe-area-context": "4.5.0",
-    "react-native-screens": "~3.20.0",
-    "react-native-tab-view": "^3.5.1"
```
Comment (mdshamoon) [inline package.json:36]: Why did we remove this?

**Review by mdshamoon — CHANGES_REQUESTED** [review]
When I am selecting the emoji for the first time in a chat. It adds undefined in the text.

File: package.json (hunk 1 above)
Comment (nishant25062002) [reply package.json:36]: Sorry I removed due to some confusion.

File: package-lock.json
Comment (nishant25062002) [reply package-lock.json:1]: I have removed.

File: components/messages/ChatInput.tsx [hunk 2]
```diff
@@ -95,6 +118,14 @@ const ChatInput: React.FC<ChatInputProps> = ({ contact }) => {
             <FontAwesome name="send" style={styles.sendicon} />
           </Pressable>
         </View>
+        {showEmoji && (
```
Comment (ajaman190) [inline components/messages/ChatInput.tsx:121]: https://github.com/glific/mobile/assets/84832565/aea2522d-e347-4ffe-8316-48b84aa053aa

1. As you can see, currently, multiple states are active at the same time. Can you create a single tab in which the content changes according to the state (emojis, attachments, options), similar behavior you can observe in WhatsApp.
2. When the emoji tab is open, replace the emoji button with a keyboard button. which on press the switch to the device keyboard and close the emoji tab.

**Review by ajaman190 — CHANGES_REQUESTED** [review]

File: components/messages/ChatInput.tsx (hunk 2 above)
Comment (Abro0058T) [reply components/messages/ChatInput.tsx:121]: @nishant25062002  ig you can make the update faster by making the whole input part as a separate component . When ever we type or in this case select an emoji the whole ChatInput component is rendering again and ig this is why the emoji update is slow . Besides u have used  8-10 libraries for this issue but none of them quick enough , the one which is quick is making a significant change in UX .

**Review by mdshamoon — APPROVED** [review]
Please resolve merge conflicts and merge after that.

**Pull request closed without merging**